pkg/route/secret/secret_monitor_test.go

```diff
@@ -0,0 +1,265 @@
+package secret
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestAddSecretEventHandler(t *testing.T) {
+	var (
+		namespace  = "ns"
+		secretName = "secret"
+	)
+
+	scenarios := []struct {
+		name              string
+		handler           cache.ResourceEventHandler
+		numInvocation     int
+		expectNumHandlers int32
+		expectKey         ObjectKey
+		expectErr         int
+	}{
+		{
+			name:              "nil handler is provided",
+			handler:           nil,
+			numInvocation:     1,
+			expectNumHandlers: 0,
+			expectErr:         1,
+		},
+		{
+			name:              "correct handler is provided",
+			handler:           cache.ResourceEventHandlerFuncs{},
+			numInvocation:     5,
+			expectKey:         NewObjectKey(namespace, secretName),
+			expectNumHandlers: 5,
+			expectErr:         0,
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			fakeKubeClient := fake.NewSimpleClientset()
+			fakeInformer := func() cache.SharedInformer {
+				return fakeSecretInformer(context.TODO(), fakeKubeClient, namespace, secretName)
+			}
+			sm := secretMonitor{
+				kubeClient: fakeKubeClient,
+				monitors:   map[ObjectKey]*monitoredItem{},
+			}
+
+			gotErr := 0
+			for i := 0; i < s.numInvocation; i++ {
+				if _, err := sm.addSecretEventHandler(context.TODO(), namespace, secretName, s.handler, fakeInformer); err != nil {
+					gotErr += 1
+				}
+			}
+			if gotErr != s.expectErr {
+				t.Errorf("expected %d errors, got %d errors", s.expectErr, gotErr)
+			}
+
+			if s.expectErr == 0 {
+				if _, exist := sm.monitors[s.expectKey]; !exist {
+					t.Fatal("monitor key should be added into map", s.expectKey)
+				}
+				if sm.monitors[s.expectKey].numHandlers.Load() != s.expectNumHandlers {
+					t.Errorf("expected %d handlers, got %d handlers", s.expectNumHandlers, sm.monitors[s.expectKey].numHandlers.Load())
+				}
```

Comment:
```suggestion
				if sm.monitors[s.expectKey].numHandlers.Load() != len(s.keys) {
					t.Errorf("expected %d handlers, got %d handlers", len(s.keys), sm.monitors[s.expectKey].numHandlers.Load())
				}
				for _, k  := range s.keys {
				         if _, exist := sm.monitors[k]; !exist {
					      t.Fatalf("expected key not found: %v", k)
					}
				}
```

Reply:
Updated it based on the changes in the fields and logic.